Source/options.cpp

```diff
@@ -812,6 +818,75 @@ void OptionEntryResampler::UpdateDependentOptions() const
 #endif
 }
 
+OptionEntryAudioDevice::OptionEntryAudioDevice()
+    : OptionEntryListBase("Device", OptionEntryFlags::CantChangeInGame, N_("Device"), N_("Audio device"))
+{
+}
+void OptionEntryAudioDevice::LoadFromIni(string_view category)
+{
+	char deviceStr[100];
+	GetIniValue(category, key, deviceStr, sizeof(deviceStr), "");
+	deviceName_ = deviceStr;
+}
+
+void OptionEntryAudioDevice::SaveToIni(string_view category) const
+{
+#if SDL_VERSION_ATLEAST(2, 0, 0)
+	SetIniValue(category, key, deviceName_);
+#endif
+}
+
+size_t OptionEntryAudioDevice::GetListSize() const
+{
+#if SDL_VERSION_ATLEAST(2, 0, 0)
+	return SDL_GetNumAudioDevices(false) + 1;
+#else
+	return 1;
+#endif
+}
+
+string_view OptionEntryAudioDevice::GetListDescription(size_t index) const
+{
+	constexpr size_t MaxChars = 32;

```

Comment:
Considering that western symbols are not all the same width maybe it would be better to truncate based on width rather then symbol count?

Reply:
Well, here's how that looks with a max width of 500.

![image](https://user-images.githubusercontent.com/9203145/171426632-0dacb24f-8724-46f4-8f07-ea195efa318d.png)
![image](https://user-images.githubusercontent.com/9203145/171426660-a50110c6-bd36-4e9e-8296-5583a5ece236.png)